source.lua

```diff
@@ -657,6 +657,16 @@ do
 			
 			close()
 		end)
+		
+		if not callback then
+			notification.Decline.Visible = false
+			notification.Accept.Visible = false
+			end
+		
+		if timer > 0 then
+			wait(timer)
```

Comment:
The yield should be on another thread as it should not halt the current thread's execution.